flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBStateBackend.java

```diff
@@ -454,9 +460,21 @@ public CheckpointStorage createCheckpointStorage(JobID jobId) throws IOException
 		LocalRecoveryConfig localRecoveryConfig =
 			env.getTaskStateManager().createLocalRecoveryConfig();
 
+		ClassLoader userClassLoader = env.getUserClassLoader();
+
+		if (optionsFactory instanceof UserConfiguredOptionsFactory) {
```

Comment:
If we would `Class.forName(...)` in the `backend.configure(...)`, this would not be required.